modules/install.js

```diff
@@ -1,67 +1,68 @@
-import { throwInvariant } from './utils';
-import { loopPromiseCaughtError } from './errors';
+import { throwInvariant } from './utils'
+import { loop, getCmd, getModel, isLoop, liftState } from './loop'
+import Cmd, { cmdToPromise, isCmd } from './cmd'
 
-import {
-  loop,
-  getEffect,
-  getModel,
-  isLoop,
-  liftState,
-} from './loop';
 
-import {
-  batch,
-  none,
-  isEffect,
-  effectToPromise,
-} from './effects';
+const noCmdPromise = Promise.resolve()
+
 
-/**
- * Installs a new dispatch function which will attempt to execute any effects
- * attached to the current model as established by the original dispatch.
- */
 export function install() {
   return (next) => (reducer, initialState, enhancer) => {
-    let currentEffect = none();
-    const [initialModel, initialEffect] = liftState(initialState);
+    const [initialModel, initialCmd] = liftState(initialState)
+    let cmdsQueue = []
 
     const liftReducer = (reducer) => (state, action) => {
-      const result = reducer(state, action);
-      const [model, effect] = liftState(result);
-      currentEffect = effect;
-      return model;
-    };
+      const result = reducer(state, action)
+      const [model, cmd] = liftState(result)
+      cmdsQueue.push({ originalAction: action, cmd })
+      return model
+    }
+
+    const store = next(liftReducer(reducer), initialModel, enhancer)
+
+    const runCmds = (queue) => {
+      const promises = queue.map(runCmd).filter((x) => x)
+      if (promises.length === 0) return noCmdPromise
+      else if (promises.length === 1) return promises[0].then(() => {})
```

Comment:
i'm not sure what the promise should actually resolve at this point so i've left it blank for now